src/main/java/io/fabric8/maven/docker/wait/ExitCodeChecker.java

```diff
@@ -0,0 +1,35 @@
+package io.fabric8.maven.docker.wait;
+
+import io.fabric8.maven.docker.access.DockerAccessException;
+import io.fabric8.maven.docker.service.ServiceHub;
+
+public class ExitCodeChecker implements WaitChecker {
+
+    private final int exitCode;
+    private final ServiceHub hub;
+    private final String containerId;
+
+    public ExitCodeChecker(int exitCode, ServiceHub hub, String containerId) {
+        this.exitCode = exitCode;
+        this.hub = hub;
+        this.containerId = containerId;
+    }
+
+    @Override
+    public boolean check() {
+        try {
+            Integer exitCode = hub.getQueryService().getMandatoryContainer(containerId).getExitCode();
```

Comment:
Please don't use a local variable with the same name as a field. Might be ok for a constructor but please not in a method.